code-challenges.test.js

```diff
@@ -47,21 +86,59 @@ const studyMinutesWeek2 = {
   friday: 15,
   saturday: 65
 }
-// Expected output: [10, 15, 20, 45, 60, 65, 100]
+// // Expected output: [10, 15, 20, 45, 60, 65, 100]
 
-// b) Create the function that makes the test pass.
 
-// --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+// // b) Create the function that makes the test pass.
 
-// a) Create a test with expect statements for each of the variables provided.
+describe ('leastToGreatest function'), () => {
+  it ('should return the least to greatest', () => {
+    expect(leastToGreatest(studyMinutesWeek1)).toEqual([15, 15, 20, 30, 30, 60, 90])
+    expect(leastToGreatest(studyMinutesWeek2)).toEqual([10, 15, 20, 45, 60, 65, 100])
+  })  
+}
+
+// // --------------------3) Create a function that takes in an array and returns an array of the accumulating sum. An empty array should return an empty array.
+
+// Pseudo Code:
+// input: a function that takes an array of numbers
+  // use the reduce method to compute the sum up until the current indexed value and is repeated through iteration until all values are done
+  // each sum should still be shown
+      // output: an array of the accumulating sum
+// if the array is empty
+      // output: an empty array
+// call the function to ensure it is working
+
+const sumOfAll = (array) => {
+  if (array.length === 0) {
+    return []
+    }
+    return array.reduce((sum, num) => {
+      return [...sum, num + sum[sum.length - 1]]
+      }, [])
+      }
```

Comment:
I see the thought process, but what happens on the first iteration? `sum.length` is 0, since the starting point of the accumulator is an empty array. Also, you are not setting `sum` to a new value on each iteration. Great effort though, love that you are trying to use reduce!